File: SLS-scripts/Oct24_testlft.m
P = Oct02_SISO_model();
K = tf([1], [1, 6, 5]);
P = ss(P);
K = ss(K);

pk = lft(P, K);

A = P.A;
B = P.B;
C = P.C;
D = P.D;

Ak = K.A;
Bk = K.B;
Ck = K.C;
Dk = K.D;

Alft = pk.A;
Blft = pk.B;
Clft = pk.C;
Dlft = pk.D;

save('Oct24_testlft_data.mat', 'A', 'B', 'C', 'D', 'Ak', 'Bk', 'Ck', 'Dk', 'Alft', 'Blft', 'Clft', 'Dlft')

[mag, phase] = bode(pk, [1])

File: SLS-scripts/Oct24_testlft.py
import control as co
import scipy.io as io

data = io.loadmat('Oct24_testlft_data.mat')

P = co.ss(data['A'], data['B'], data['C'], data['D'])
K = co.ss(data['Ak'], data['Bk'], data['Ck'], data['Dk'])
pk = co.ss(data['Alft'], data['Blft'], data['Clft'], data['Dlft'])

pk_co = P.lft(K)
mag, phase, _ = pk_co.freqresp([1])
import IPython
if IPython.get_ipython() is None:
    IPython.embed()
